pandas/tseries/tools.py

```diff
@@ -271,6 +285,14 @@ def to_datetime(arg, errors='raise', dayfirst=False, yearfirst=False,
     >>> pd.to_datetime('13000101', format='%Y%m%d', errors='coerce')
     NaT
     """
+    
+    if origin is not None:
+        from pandas.core.api import Timestamp
+        if origin == 'julian':
+            arg = arg - Timestamp(0).to_julian_date()
+        else:
+            arg = arg + Timestamp(origin).to_julian_date() - Timestamp(0).to_julian_date()
```

Comment:
I want to see some more actual usecases. All of the tests are just variations on a single date.